app/forms/register_user_email_form.rb

```diff
@@ -29,6 +29,21 @@ def email
     email_address&.email
   end
 
+  def email_fingerprint
+    email_address&.email_fingerprint
+  end
+
+  def normalized_email(email)
+    return @normalized_email if defined?(@normalized_email)
+
+    @normalized_email = email.blank? ? email.to_s : EmailNormalizer.new(
+      email,
+      google_mx_lookup: false,
+    ).normalized_email
+  rescue Mail::Field::IncompleteParseError
+    email
+  end
```

Comment:
so I just realized we are already doing an MX record check/email normalization inside this form. It's "hidden" inside the `blocked_email_address`s `SuspendedEmail.find_with_email(email)` implementation. Maybe there's a way we can refactor that path so that we build the normalized email once and then do different things with the results?